docs/gitbook/reference/interbit-middleware/README.md

```diff
@@ -1,9 +1,159 @@
 # Interbit Middleware
 
+Redux middleware to connect configured interbit blockchains to redux store state.
 
+This middleware creates a connected interface and a hypervisor, then loads and connects to chains and peers that have been specified in the index.html file.
 
+## Setting Up
 
-<div class="tips danger">
+### Installing dependencies
+
+The middleware is included in the `interbit-ui-tools` package.
+
+```sh
+npm i --save interbit-ui-tools redux redux-saga
+```
+
+### Creating the Redux Store
+
+Because the interbit middleware contains a saga that must be run there are several steps to including it in your redux store.
+
+1. Creating the Interbit middleware
+1. Creating the Redux Saga middleware
+1. Including the reducer
+1. Including the middlewares with your store on `createStore`
+1. Running the saga
+
+```js
+import {
+  combineReducers,
+  createStore,
+  applyMiddleware
+} from 'redux'
+
+import {
+  reducer as interbitReducer,
+  createMiddleware as createInterbitMiddleware,
+  rootSaga as interbitSaga
+} from 'interbit-ui-tools'
+
+import createSagaMiddleware from 'redux-saga'
+
+
+const createStore = () => {
+  // 1,2. Creating the middlewares
+  const interbitMiddleware = createInterbitMiddleware()
+  const sagaMiddleware = createSagaMiddleware()
+
+  // 3. Including the reducers
+  const reducers = combineReducers({
+    interbit: interbitReducer,
+    // ... other reducers
+  })
+
+  // 4. Including the middlewares
+  const store = createStore(
+    reducers,
+    applyMiddleware(interbitMiddleware, sagaMiddleware)
+  )
+
+  // 5. Running the saga
+  sagaMiddleware.run(interbitSaga)
+
+  return store
+}
+```
+
+### Specifying Chains in index.html
+
+The middleware automatically connects to any chains that are specified on the first DOM element in index.html with an id of `"interbit"`.
+
+Specify chain IDs using a data tag starting with data-chain-id-chain-name. The chain-name will be camel case when it is used as a chain alias to mount state on in the middleware's redux store.
+
+```html
+<div id="interbit" data-chain-id-my-chain="32b9365f325da189439f6d453593cf4e0a5a06ead7e6bbecc1d0814c578df452" />
+```
+
+The above data tag will result in a chain named `myChain` being mounted in the middleware state as follows:
+
+```js
+const state = {
+  interbit: {
+    chains: {
+      myChain: { ...chainState }
+    }
+  }
+}
+```
+
+### Specifying peers in index.html
+
+You can specify peers for your browser node to connect to in index.html as well. There are specified the same was as chains using the data-peer-hints attribute.
+
```

Comment:
Need to be a little more specific. Need to be specified as 'localhost:8080' or 'localhost' (no protocol)

Reply:
This is specified in the next line.